### PR TITLE
peatanktest

### PR DESCRIPTION
**Pull request by zhouxs22**

### CODE REVIEW

**Comment by qinchuanhui** [comment]
Have you finish it or just a test?  Please add some description about your work.

File: src/battle_game/app/app.cpp [hunk 1]
```diff
@@ -451,3 +451,4 @@ void App::UpdateImGui() {
   ImGui::Render();
 }
 }  // namespace battle_game
+// pull request test
```
Comment (qinchuanhui) [inline src/battle_game/app/app.cpp:454]: Unrealeted file changes like this (or something in `fonts` folder), should not be included to commit and push.

**Comment by zhouxs22** [comment]
My works:
1 add pea_tank.cpp and pea_tank.h in src/battle_game/core/units which is designed to shoot in 4 directions simutaneously with each bullet dealing half damage
2 add speedbullet:(defined in src/battle_game/core/units) which is designed to increase its speed after some certain number of updates(my solution to issue#296)
These works are NOT TESTED.( I spent about half a week searching on Google/Stack Overflow and asking TAs/classmates/in the WeChat Group but still cannot get my Visual Studio generating battle.exe correctly, so I cannot do the test temporarily.I'll try to work on it in the remaining days but please grade it if I cannot get Visual Studio work before the deadline.)

**Comment by qinchuanhui** [comment]
plz fix your bugs first

**Comment by zhouxs22** [comment]
> plz fix your bugs first

It seems they're fixed now. Sorry about that since my VSCode seemed to be not working well (and both me and TA & classmates & Google I sought for help couldn't really fix) and it may not build successfully even when there was no bugs at all (and GitHub ran the version of code successfully)

File: src/battle_game/core/bullets/speed_bullets.h [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+               float rotation,
+               float damage_scale,
+               glm::vec2 velocity);
+  ~SpeedBullets() override;
+  void Render() override;
+  void Update() override;
+
+ private:
+  glm::vec2 velocity_{};
+  float count;
```
Comment (qinchuanhui) [inline src/battle_game/core/bullets/speed_bullets.h:21]: naming: `count_`

**Review by qinchuanhui — CHANGES_REQUESTED** [review]
Don't submit your `.obsidian` folder, just delete it from ur code-repo

**Comment by zhouxs22** [comment]
> Don't submit your `.obsidian` folder, just delete it from ur code-repo

'.obsidian' deleted and 'count' variable renamed as 'count_'

**Review by qinchuanhui — APPROVED** [review]
LGTM, you can get the corresponding score

**Pull request closed without merging**